0.29 CLI Documentation (#762)
astronomer/docs: software/cli-reference.md

brew install astronomer/tap/astro
```

> To install a specific version of the Astro CLI, you'll have to specify `@major.minor.patch`. To install v0.16.1, for example, run:

Reviewer: ```suggestion
To install a specific version of the Astro CLI, specify `@major.minor.patch`. To install v0.28.1, for example, run:
```

```sh

> brew install astronomer/tap/astro@0.16.1

Reviewer: ```suggestion
brew install astronomer/tap/astro@0.28.1
```

```

### Install with cURL

To install the latest version of the Astro CLI, run:

```
curl -sSL https://install.astronomer.io | sudo bash
```

To install a specific version of the Astro CLI, specify `-s -- major.minor.patch` as a flag at the end of the cURL command. To install v0.16.1, for example, run:

```

> curl -sSL https://install.astronomer.io | sudo bash -s -- v0.16.1

Reviewer: ```suggestion
curl -sSL https://install.astronomer.io | sudo bash -s -- v0.28.1
```

If the installation was successful, you should see the version of the CLI that you installed in the output:

```

> Astro CLI Version: 0.15.0

Reviewer: ```suggestion
Astro CLI Version: 1.0.0
```

Once you've successfully installed the CLI, use the remainder of this guide to learn more about the CLI's available commands.

> ## astro auth

Reviewer 2: 0.29 is a continuation of 0.28, and does not have breaking changes of 1.0.0, so both `astro auth` & `astro cluster` command remain as it is

The settings that you can update via the command line are:

- `cloud.api.protocol`
- `cloud.api.port`
- `cloud.api.ws_protocol`
- `cloud.api.token`
- `container.engine`
- `context`
- `contexts`
- `houston.dial_timeout`
- `local.houston`
- `local.orbit`
- `postgres.user`
- `postgres.password`
- `postgres.host`
- `postgres.port`
- `project.deployment`
- `project.name`
- `project.workspace`
- `webserver.port`
- `show_warnings`

### Subcommands

| Subcommand | Usage                                                     |
| ---------- | --------------------------------------------------------- |
| `get`      | Show current values for the above configuration settings. |
| `set`      | Updates a setting in your platform to a new value.        |

### Related documentation

- [Apply a Platform Configuration Change on Astronomer](apply-platform-config.md)

> ## astro context delete

Reviewer 2: same reasoning as above, `astro context` command won't be present in 0.29

- [Deployment Logs](deployment-logs.md)

## astro deployment runtime migrate

Migrate an existing Software Deployment from Astronomer Certified to Astro Runtime

### Usage

`astro deployment runtime migrate --deployment-id=<deployment-id>`

### Flags

| Flag       | Value Type                                    | Usage                                                               |
| ---------- | --------------------------------------------- | ------------------------------------------------------------------- |
| `--cancel` | None                                          | Cancel the migration.                                      |

> | `--deployment-id` | String                                        | The Deployment that you want to migrate. |

Reviewer: ```suggestion
| `--deployment-id` (Required) | String                                        | The Deployment that you want to migrate. |
```

> | `--deployment-id` | String                                        | The Deployment that you want to migrate. |

Reviewer 2: not sure if we would need to fix indentation of the table

## astro deployment runtime update

Update an existing Software Deployment's Astro Runtime version

### Usage

> `astro deployment runtime upgrade --deployment-id=<deployment-id> `

Reviewer: ```suggestion
`astro deployment runtime update --deployment-id=<deployment-id> `
```

Is it upgrade or update?

### Flags

| Flag       | Value Type                                    | Usage                                                               |
| ---------- | --------------------------------------------- | ------------------------------------------------------------------- |
| `--cancel` | None                                          | Cancel the migration.                                      |

> | `--deployment-id` | String                                        | The Deployment that you want to migrate. |

Reviewer: ```suggestion
| `--deployment-id` (Required) | String                                        | The Deployment that you want to migrate. |
```

> | `--desired-runtime-version` | String                                        | The Astro Runtime version you want to upgrade to. |

Reviewer: ```suggestion
| `--desired-runtime-version` (Required) | String                                        | The Astro Runtime version you want to upgrade to. |
```

> | `--desired-runtime-version` | String                                        | The Astro Runtime version you want to upgrade to. |

Reviewer 2: not sure if we would need to fix indentation of the table

### Flags

| Flag                        | Value Type | Usage                                                                                                                                                                                         |
| --------------------------- | ---------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--deployment-id`(Required) | String     | The Airflow Deployment in which the service account is configured. Use this flag as an alternative to specifying `<your-service-account-id>`. To get this value, run `astro deployment list`. |

### Related documentation

- [Deploy to Astronomer via CI/CD](ci-cd.md)

> ## astro deployment service-account list

Reviewer 2: the same reasoning as before, this change is not included in 0.29

### Usage

> `astro deployment user add --email=<user-email-address> --deployment-id=<user-deployment-id> --role<user-role>`

Reviewer 2: the same reasoning as before, this change is not included in 0.29

### Flags

| Flag                         | Value Type | Usage                                                                                                                                            |
| ---------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| `--deployment-id` (Required) | String     | The ID of the Deployment that the user will be added to. To find this value, run `astro deployment list`.                                        |
| `--email` (Required)         | String     | The email for the user.                                                                                                                          |
| `--role` (Required)          | String     | The role assigned to the user. Can be `DEPLOYMENT_VIEWER`, `DEPLOYMENT_EDITOR`, or `DEPLOYMENT_ADMIN`. The default value is `DEPLOYMENT_VIEWER`. |

### Related documentation

- [Manage User Permissions on Astronomer](workspace-permissions.md)

> ## astro deployment user remove

Reviewer 2: the same reasoning as before, this change is not included in 0.29

- [Running the Airflow Upgrade Check Package](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html#upgrade-check)

> ## astro login/ logout

Reviewer 2: the same reasoning as before, this command change is not included in 0.29

Logs you in and out of an installation on Astronomer Software.

### Usage

Run `astro login <base-domain>` or `astro logout <base-domain>` to log in or out of your Astronomer platform respectively. This is equivalent to using the login screen of the Software UI.

If you have access to more than one Astronomer installation, each will have a unique `<base-domain>`. When switching between platforms, make sure to log out of one `<base domain>` before logging into another.

> 

Reviewer: These should be 2 separate entries no?

### Usage

> `astro workspace create --name=<new-workspace-name> [flags]`

Reviewer 2: the same reasoning as before, this change is not included in 0.29

### Flags

| Flag                  | Value Type | Usage                                  |
| --------------------- | ---------- | -------------------------------------- |

> | `--label` (_required_) | String     | The label/name for the new Workspace.        |

Reviewer: ```suggestion
| `--name` (_required_) | String     | The name for the new Workspace.        |
```

Changed to name?

### Flags

> | Flag                        | Value Type | Usage                                                                                                        |

Reviewer 2: the same reasoning as before, this change is not included in 0.29

| --------------------------- | ---------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--workspace-id` (Required) | String     | The Workspace you're creating a service account for.                                                                                                 |

> | `--label` (Required)        | String     | A label for the service account.                                                                                                                     |

Reviewer: Is this name now? If so make sure to update all other Workspace commands

### Flags

> | Flag        | Value Type | Usage                            |

Reviewer 2: the same reasoning as before, this change is not included in 0.29

### Usage

> `astro workspace user add [flags] <user-email-address>`

Reviewer 2: the same reasoning as before, this change is not included in 0.29

`astro workspace user add --email <user-email-address> [flags]`

### Flags

> | Flag                        | Value Type | Usage                                                                                                                                                                     |

Reviewer 2: the same reasoning as before, this change is not included in 0.29

### Usage

> `astro workspace user remove <user-email-address>`

Reviewer 2: the same reasoning as before, this change is not included in 0.29

| `--workspace-id` | String     | The Workspace that you're searching in. Specify this flag if you want to search for users in a Workspace other than your current Workspace. |
| `--email`        | String     | The email for the user you're searching for.                                                                                                |
| `--name`         | String     | The name of the user to search for.                                                                                                         |

> | `--user-id`      | String     | The ID of the user to search for.                                                                                                           |

Reviewer 2: the same reasoning as before, this change is not included in 0.29

### Usage

> `astro workspace user update --email <user-email-address> [flags]`

Reviewer 2: the same reasoning as before, this change is not included in 0.29
